projects/packages/my-jetpack/src/class-rest-products.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Sets up the Products REST API endpoints.
+ *
+ * @package automattic/my-jetpack
+ */
+
+namespace Automattic\Jetpack\My_Jetpack;
+
+require_once 'class-products.php';
+
+/**
+ * Registers the REST routes for Products.
+ */
+class REST_Products {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		register_rest_route(
+			'my-jetpack/v1',
```

Comment:
@fgiannar what do you think about this?

Reply:
If I'm not wrong, one of the goals of My Jetpack is to be portable independently of other packages, plugins. If so, it needs to implement its own endpoints and also, with its own namespace in order to avoid collisions.

> we are adding all endpoints under the jetpack/v4 namespace...

If we keep the same `jetpack/v4` namespace, we will have to start to name these new endpoints with something that guarantees no names collision, which could end up with some kind of weird names.

[/jetpack/v4/backup-capabilities](https://github.com/Automattic/jetpack/blob/ad487f4d78ad1d1fe237819772246fc1cb03ec19/projects/plugins/backup/src/php/class-jetpack-backup.php#L153), [/backup-promoted-product-info'](https://github.com/Automattic/jetpack/blob/ad487f4d78ad1d1fe237819772246fc1cb03ec19/projects/plugins/backup/src/php/class-jetpack-backup.php#L177) are some examples where we had to add the `backup` word into the endpoint name. Probably, I'd suggest something like `/backup/v1/capabilities` and `/backup/v1/promoted-product-info`.

What's the issue of using `my-jetpack/v1` as a namespace?

Reply:
I don't see any immediate issue with using `my-jetpack/v1` as namespace. The reason we kept the `jetpack/v4` namespace for other endpoints used in latest packages is mainly backwards compatibility as we were moving those endpoints from Jetpack-the-plugin to those packages rather than creating new ones. 
That being said I'd probably go for `jetpack/v4` for consistency.

@jeherve Are you aware of any plugins that could be whitelisting Jetpack REST endpoints using the `jetpack/v4` namespace?

Reply:
I'm sorry but if there are no issues with `my-jetpack/v1` namespace, if it isn't a matter of backward compatibility issues either, why do we need to get tied to `jetpackv4` namespace? Most of these endpoints are defined by the Jetpack plugin, and if I'm not wrong, the spirit of the Jetpack dev tool is to try to provide independent packages,  plugins to each other.
In the same way that I wouldn't use `wp/v2` namespace when adding customs endpoint, I wouldn't use `jetpack/v4` either, beyond we (still) are using the same namespace among all packages/plugins that belong to the Jetpack ecosystem.
My two 🪙 s

Reply:
Yes, let's move ahead this way.

Reply:
Thanks, Leo. We can re-define the names in case we realize `my-jetpack` is a very bad idea. 💡